libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -236,14 +240,15 @@ def fmt_err(x: str) -> str:
 
 
 class AzureSearch(VectorStore):
-    """`Azure Cognitive Search` vector store."""
+    """`Azure AI Search` vector store."""
 
     def __init__(
         self,
         azure_search_endpoint: str,
-        azure_search_key: str,
-        index_name: str,
         embedding_function: Union[Callable, Embeddings],
+        azure_search_key: str = "",
```

Comment:
If the parameter order changed, that'd be a backwards compatible issue for library users, right?

Reply:
It would be. My problem was getting `azure_search_key: Optional[str]` to work here. I'd get an error unless I explicitly added the default. My ideal would be to not have to. Do you have any thoughts on how to fix this? I might try refactor in a bit

Reply:
Let me actually change this into an if else statement and change the order to first check for the token and the else out to api key.

Reply:
@pamelafox adding a lambda as the default for embeddings and then maintaining the order of parameters worked 😄 Mypy is happy and the code should be backwards compatible now!